ambry-frontend/src/test/java/com/github/ambry/frontend/FrontendQuotaIntegrationTest.java

```diff
@@ -0,0 +1,508 @@
+import com.github.ambry.account.Container;
+import com.github.ambry.account.InMemAccountService;
+import com.github.ambry.account.InMemAccountServiceFactory;
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.clustermap.Partition;
+import com.github.ambry.clustermap.PartitionState;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.commons.CommonTestUtils;
+import com.github.ambry.commons.LoggingNotificationSystem;
+import com.github.ambry.commons.NettySslFactory;
+import com.github.ambry.commons.SSLFactory;
+import com.github.ambry.commons.TestSSLUtils;
+import com.github.ambry.config.FrontendConfig;
+import com.github.ambry.config.NettyConfig;
+import com.github.ambry.config.QuotaConfig;
+import com.github.ambry.config.SSLConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.quota.QuotaMode;
+import com.github.ambry.rest.NettyClient;
+import com.github.ambry.rest.RestServer;
+import com.github.ambry.rest.RestServiceException;
+import com.github.ambry.rest.RestUtils;
+import com.github.ambry.router.ByteRange;
+import com.github.ambry.utils.TestUtils;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpUtil;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Integration tests for Ambry frontend with quota.
+ */
+@RunWith(Parameterized.class)
+public class FrontendQuotaIntegrationTest extends FrontendIntegrationTestBase {
+  private static final int PLAINTEXT_SERVER_PORT = 1174;
+  private static final int SSL_SERVER_PORT = 1175;
+  private static final int MAX_MULTIPART_POST_SIZE_BYTES = 10 * 10 * 1024;
+  private static final String DEFAULT_PARTITION_CLASS = "default-partition-class";
+  private static final MockClusterMap CLUSTER_MAP;
+  private static final VerifiableProperties FRONTEND_VERIFIABLE_PROPS;
+  private static final File TRUST_STORE_FILE;
+  private static final FrontendConfig FRONTEND_CONFIG;
+  private static final InMemAccountService ACCOUNT_SERVICE =
+      new InMemAccountServiceFactory(false, true).getAccountService();
+  private static Account ACCOUNT;
+  private static Container CONTAINER;
+  private static boolean enableUndeleteTested = false;
```

Comment:
if undelete is always enabled, we can remove this.

Reply:
done.